### ISSUE TITLE
Custom Schema name

### ISSUE THREAD

**Issue by mohpor** [mczachurski/Swiftgger#4]
Hi,

First off, wonderful package!

I'm trying to add my models to schemas for the swagger specs. We are using `enum`s to encapsulate models and functionalities, therefore, my model names are not exactly clear when added as `APIObject`.

For example this might be a model of mine:
```swift
enum Models {
  enum Transactions {
    enum Response {
    final class List: Content {
       ....
    }
  }
 }
}
```

now, as you can see, when I add a transaction list item, all I get in Schemas section is: `List` which is not at all clear to user. 
Is there something I'm missing?

My suggestion (Swift being crazy when it comes to reflection, and use-cases being very different), it might be a good idea to let developers choose the type's name when adding the `APIObject`.

Thanks

**Comment by mohpor** [comment]
@mczachurski is it possible then?

**Comment by mczachurski** [comment]
Hi,

Your approach with enums is very interesting. I didn't realize that this kind of construction in Swift is even possible :-) (a little bit like namespaces in other languages). I veryfied that in Swagger it's possible to have object names with dots like: `Models.Transactions.Response.List`. 

Of course it's possible to add additional parameter to `APIObject`:

```swift
APIObject(object: Models.Transactions.Response.UserDto(id: UUID(), name: "John Doe"), withCustomName: "Models.Transactions.Response.UserDto")
```

However that parameter will be a string so it will be harder to maintain (refactoring will be impossible here).

Then we need to change process of generating `OpenAPIDocument` becouse now it's really easy to connect objects from requests/responses with schamas (class name from reflection is a key):

```swift
/// 'UserDto' is a link to schemas dictionary.
APIResponse(code: "200", description: "Specific item", object: UserDto.self)

/// 'UserDto' is a key in schemas dictionary.
APIObject(object: UserDto(id: UUID(), name: "John Doe"))
```

That change will require some more work.

**Comment by mohpor** [comment]
@mczachurski 
I will prefer it that way too, you're right, that's more maintainable.

(Regarding my approach, yes it is possible and very much needed for me as SwiftLint might throw tantrums sometime, I Package my models and Static functionality inside case-less `enum`s because they can not be initialized on their own.) 

One more thing to add is, if I have a type associated with a variable in a class, it is resolved to it's full identifier (I mean Models.XYZ.Root.Item...), which means it is possible to resolve the Type to its full extent.

**Comment by mczachurski** [comment]
If you know how to use reflection in Swift to get full class name (in your case it will be `Models.Transactions.Response.List`) please let me know. Then this change should be really easy. I cannot find any solution for that.

**Comment by mohpor** [comment]
Ok, I'll try to do a PR then.

**Comment by mohpor** [comment]
@mczachurski 
I have created a Pull Request (#5 ) that addresses this problem. I have tested it and it works great.
Would love to see it in the repo ASAP.

**Comment by eneko** [comment]
@mczachurski, @mohpor, please see my notes in #5. A priori, the code looks good and seems to work well, however, it might not be a good idea to expose namespace models like that.

**Comment by mohpor** [comment]
@eneko I'm in total agreement with you, that's why I proposed to have custom name in `APIObject` constructor. It makes maintenance a little more cumbersome, but at the end of the day, there should be no Forced constraints on what a client names the internal model based on an API (unlike protobuf for example).
I have explained my case, proposed and implemented a solution (however unsuitable) and the rest is up to @mczachurski.
